spec/system/admin/product_import_spec.rb

```diff
@@ -240,16 +240,21 @@
     end
 
     it "can reset product stock to zero for products not present in the CSV" do
+      pending("#12933")
+
       csv_data = <<~CSV
         name, producer, category, on_hand, price, units, unit_type, shipping_category_id
         Carrots, User Enterprise, Vegetables, 500, 3.20, 500, g, #{shipping_category_id_str}
       CSV
       File.write('/tmp/test.csv', csv_data)
 
+      # setting a variant to have negative stock, with the on demand option set to true
+      Spree::Product.find_by(name: 'Cabbage').variants.first.on_demand = true
+      Spree::Product.find_by(name: 'Cabbage').variants.first.on_hand = -30
+
```

Comment:
I could just do one query :
```suggestion
     # setting a variant to have negative stock, with the on demand option set to true
      cabbage_variant = Spree::Product.find_by(name: 'Cabbage').variants.first
      cabbage_variant.on_demand = true
      cabbage_variant.on_hand = -30
```

Reply:
Agreed. another way to refer is with the existing variable `product4.variants.first` but this is good.

Reply:
Thanks @rioug for the review and @dacook for committing, apologies I've missed this review in useful time.